account_move_helper/models/account_account.py

```diff
@@ -19,34 +18,39 @@ class AccountAccount(models.Model):
         inverse='_inverse_new_balance'
     )
 
-    @api.multi
     @api.depends('balance')
     def _compute_new_balance(self):
         move_id = self._context.get('active_id', False)
         if not move_id:
             return False
+        AccountMoveLine = self.env['account.move.line']
         for rec in self:
-            move_line = self.env['account.move.line'].search([
+            move_line = AccountMoveLine.search([
                 ('move_id', '=', move_id),
                 ('account_id', '=', rec.id)], limit=1)
-            rec.new_balance = rec.balance + move_line.balance
+            rec.update({'new_balance': rec.balance + move_line.balance})
 
     @api.multi
     def _compute_balance(self):
+        move_id = self._context.get('active_id', False)
+        move = self.env['account.move'].browse(move_id)
         for rec in self:
-            # rec.balance = rec._get_balance()
+            domain = [
+                ('account_id', '=', rec.id),
+                ('move_id.state', '=', 'posted'),
+            ]
+            if move:
+                domain.append(('date', '<=', move.date))
             rec.balance = sum(
-                rec.env['account.move.line'].search([
-                    ('account_id', '=', rec.id),
-                    ('move_id.state', '=', 'posted'),
-                ]).mapped('balance'))
+                rec.env['account.move.line'].search(domain).mapped('balance'))
```

Comment:
Hi Nico! How are you?
Here you can make a variable outside For, like this and update:
@api.multi
    def _compute_balance(self):
        move_id = self._context.get('active_id', False)
        move = self.env['account.move'].browse(move_id)
        AccountMoveLine = self.env['account.move.line']
        for rec in self:
            domain = [
                ('account_id', '=', rec.id),
                ('move_id.state', '=', 'posted'),
            ]
            if move:
                domain.append(('date', '<=', move.date))
            rec.update({'balance': sum(AccountMoveLine.search(domain).mapped('balance'))})

Reply:
Hi Valentino
Thanks for the review.
I've made the changes.